WordPress/src/main/java/org/wordpress/android/datasets/ReaderDatabase.java

```diff
@@ -144,12 +146,29 @@ public void onCreate(SQLiteDatabase db) {
         createAllTables(db);
     }
 
+    @SuppressWarnings({"FallThrough"})
     @Override
     public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
-        // for now just reset the db when upgrading, future versions may want to avoid this
-        // and modify table structures, etc., on upgrade while preserving data
-        AppLog.i(T.READER, "Upgrading database from version " + oldVersion + " to version " + newVersion);
-        reset(db);
+        AppLog.i(T.READER,
+                "Upgrading database from version " + oldVersion + " to version " + newVersion + " IN PROGRESS");
+        int currentVersion = oldVersion;
+        if (currentVersion <= DB_LAST_VERSION_WITHOUT_MIGRATION_SCRIPT) {
+            // versions 0 - 135 didn't support migration scripts, so we can safely drop and recreate all tables
+            reset(db);
+            currentVersion = DB_LAST_VERSION_WITHOUT_MIGRATION_SCRIPT;
+        }
+
+        switch (currentVersion) {
+            case 135:
```

Comment:
I left it out intentionally, because I wanted to highlight that this is the switch where you should put case for each version number. Moreover `case X` is actually for migration from version `X` to version `X+1`, so the naming DB_LAST_VERSION_WITHOUT_MIGRATION_SCRIPT doesn't make much sense. I'm not sure about it, wdyt?